egui/src/widgets/plot/items.rs

```diff
@@ -73,13 +73,317 @@ struct ExplicitGenerator {
+pub struct Marker {
+    pub(crate) shape: MarkerShape,
+    /// Color of the marker. `Color32::TRANSPARENT` means that it will be picked automatically.
+    pub(crate) color: Color32,
+    /// Whether to fill the marker. Does not apply to all types.
+    pub(crate) filled: bool,
+    /// The maximum extent of the marker from its center.
+    pub(crate) radius: f32,
+}
+
+impl Default for Marker {
+    fn default() -> Self {
+        Self {
+            shape: MarkerShape::Circle,
+            color: Color32::TRANSPARENT,
+            filled: true,
+            radius: 2.0,
+        }
+    }
+}
+
+impl Marker {
+    /// Get a vector containing a marker of each shape.
+    pub fn all() -> Vec<Self> {
+        vec![
+            Self::circle(),
+            Self::diamond(),
+            Self::square(),
+            Self::cross(),
+            Self::plus(),
+            Self::up(),
+            Self::down(),
+            Self::left(),
+            Self::right(),
+            Self::asterisk(),
+        ]
+    }
+
+    pub fn circle() -> Self {
+        Self {
+            shape: MarkerShape::Circle,
+            ..Default::default()
+        }
+    }
+
+    pub fn diamond() -> Self {
+        Self {
+            shape: MarkerShape::Diamond,
+            ..Default::default()
+        }
+    }
+
+    pub fn square() -> Self {
+        Self {
+            shape: MarkerShape::Square,
+            ..Default::default()
+        }
+    }
+
+    pub fn cross() -> Self {
+        Self {
+            shape: MarkerShape::Cross,
+            ..Default::default()
+        }
+    }
+
+    pub fn plus() -> Self {
+        Self {
+            shape: MarkerShape::Plus,
+            ..Default::default()
+        }
+    }
+
+    pub fn up() -> Self {
+        Self {
+            shape: MarkerShape::Up,
+            ..Default::default()
+        }
+    }
+
+    pub fn down() -> Self {
+        Self {
+            shape: MarkerShape::Down,
+            ..Default::default()
+        }
+    }
+
+    pub fn left() -> Self {
+        Self {
+            shape: MarkerShape::Left,
+            ..Default::default()
+        }
+    }
+
+    pub fn right() -> Self {
+        Self {
+            shape: MarkerShape::Right,
+            ..Default::default()
+        }
+    }
+
+    pub fn asterisk() -> Self {
+        Self {
+            shape: MarkerShape::Asterisk,
+            ..Default::default()
+        }
+    }
+
+    /// Set the marker's color. Defaults to the curve's color.
+    pub fn color(mut self, color: Color32) -> Self {
+        self.color = color;
+        self
+    }
+
+    /// Whether to fill the marker.
+    pub fn filled(mut self, filled: bool) -> Self {
+        self.filled = filled;
+        self
+    }
+
+    /// Set the maximum extent of the marker around its position.
+    pub fn radius(mut self, radius: f32) -> Self {
+        self.radius = radius;
+        self
+    }
+
+    pub(crate) fn get_shapes(&self, position: &Pos2) -> Vec<Shape> {
```

Comment:
To reduce the number of allocations it would be better to pass in a `out_shapes: &mut Vec<Shape>` and push into that